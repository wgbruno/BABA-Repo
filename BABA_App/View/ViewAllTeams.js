import React, {useEffect, useState} from 'react';
import { Button, Text, TextInput, TouchableOpacity, View, StyleSheet } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';
import { NavigationContainer } from 'react-native';
import {createStackNavigator} from 'react-navigation-stack';
import { db } from './AddTeam';

export default function ViewAllTeams({ navigation }){
    const [items, setItems] = useState([]);
    const [empty, setEmpty] = useState([]);

    useEffect(() =>{
        db.transaction((tx) => {
            tx.executeSql(
                'SELECT * FROM Team_Table', [], (tx, results) => {
                    var temp = [];
                    for(let i = 0; i < results.rows.length; i++){
                        temp.push(results.rows.item(i));
                    }
                    setItems(temp);

                    if(results.rows.length >= 1){
                        setEmpty(false);
                    } else{
                        setEmpty(true);
                    }
                }
            );
        });
    }, []);

    const listAllTeams = () => {
        return(
            <View 
            style={{height: 1, width: '100%', backgroundColor: '#000'}}
            />
        );
    };

    const emptyDatabase = () => {
        return(
            <View style={{justifyContent: 'center', alignItems: 'center', flex: 1}}>
                <Text style={{fontSize: 25, textAlign: 'center'}}>
                    No Teams Have Registered Yet
                </Text>
            </View>
        );
    }

    const styles = StyleSheet.create({
        itemsStyle:{
            fontSize:22,
            color: '#000'
        }
    });

    return(<>
        <View style={{flex: 1}}>
            {empty ? emptyDatabase(empty):
            <FlatList
                data={items}
                ItemSeparatorComponent={listAllTeams}
                keyExtractor={(item, index) => index.toString()}
                renderItem={({ item }) =>
                <View key={item.number} style={{ padding: 20}}>
                    <TouchableOpacity 
                        onPress={() => {navigation.navigate('UpdateTeams', {
                        paramID: item.name,
                        paramFirst: item.record,
                        paramLast: item.seed,
                        paramAge: item.players,});}}>
                        <Text style={styles.itemsStyle}>Name: {item.firstName}</Text>
                        <Text style={styles.itemsStyle}>Record: {item.lastName}</Text>
                        <Text style={styles.itemsStyle}>Seed: {item.age}</Text>
                        <Text style={styles.itemsStyle}>Players: {item.number}</Text>
                    </TouchableOpacity>
                </View>
                }
            />
            }
        </View>
    </>);
}