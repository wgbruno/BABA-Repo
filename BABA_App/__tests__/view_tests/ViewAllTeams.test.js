import {expect, jest, test} from '@jest/globals';
import renderer from 'react-test-renderer';
import ViewAllTeams from '../../View/ViewAllTeams';


test("View All Teams Screen", function (){
    const temp = renderer.create(<ViewAllTeams/>).toJSON();
    expect(temp).toMatchSnapshot();
})