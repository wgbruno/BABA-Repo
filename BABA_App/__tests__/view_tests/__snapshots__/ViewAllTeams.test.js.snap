// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View All Teams Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/images/Logo_1.png",
      }
    }
    style={
      Array [
        Object {
          "maxHeight": 150,
          "maxWidth": 200,
          "width": "3000%",
        },
        Object {
          "height": 26680,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "darkblue",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 8,
      }
    }
  >
    All Players
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      No Teams Have Registered Yet
    </Text>
  </View>
</View>
`;
